nifi-api/src/main/java/org/apache/nifi/authorization/AccessPolicy.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.authorization;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Defines a policy for a set of entities to perform a set of actions on a given resource.
+ */
+public class AccessPolicy {
+
+    private final String identifier;
+
+    private final Resource resource;
+
+    private final Set<String> entities;
+
+    private final Set<RequestAction> actions;
+
+    private AccessPolicy(final AccessPolicyBuilder builder) {
+        this.identifier = builder.identifier;
+        this.resource = builder.resource;
+
+        Set<String> entities = new HashSet<>();
+        if (builder.entities != null) {
```

Comment:
Pretty sure this is guaranteed non-null based on the Builder. This comment applies to all the Set's in User, Group, and AccessPolicy.